yjit/src/codegen.rs

```diff
@@ -642,12 +642,10 @@ fn gen_check_ints(asm: &mut Assembler, side_exit: CodePtr) {
     // see RUBY_VM_CHECK_INTS(ec) macro
     asm.comment("RUBY_VM_CHECK_INTS(ec)");
 
-    let not_mask = asm.not(Opnd::mem(32, EC, RUBY_OFFSET_EC_INTERRUPT_MASK));
-
-    asm.test(
-        Opnd::mem(32, EC, RUBY_OFFSET_EC_INTERRUPT_FLAG),
-        not_mask,
-    );
+    // Not checking interrupt_mask since it's zero outside finalize_deferred_heap_pages,
+    // signal_exec, or rb_postponed_job_flush.
+    let interrupt_flag = asm.load(Opnd::mem(32, EC, RUBY_OFFSET_EC_INTERRUPT_FLAG));
+    asm.test(interrupt_flag, interrupt_flag);
```

Comment:
Small detail but, do you need to explicitly do the load, or can you let the backend do a load if it needs to?

Reply:
I didn't check it, but it doesn't seem to work as I wanted.

The current behavior is:

```nas
  # RUBY_VM_CHECK_INTS(ec)
  0x5625a0fbd060: mov eax, dword ptr [r12 + 0x20]
  0x5625a0fbd065: test eax, eax
  0x5625a0fbd067: jne 0x5625a0fbf02d
```

and if you remove `asm.load`, it becomes:

```nas
  # RUBY_VM_CHECK_INTS(ec)
  0x5594306e5060: mov eax, dword ptr [r12 + 0x20]
  0x5594306e5065: test dword ptr [r12 + 0x20], eax
  0x5594306e506a: jne 0x5594306e702d
```

I can have a look at the backend if you prefer removing `asm.load`. I'm personally okay with having `asm.load` here since there's one benefit that I can predict what's gonna happen.

Reply:
Looks good as is 👍